TestViewer/MainWindow.xaml.cs

```diff
@@ -35,6 +35,15 @@ private void LoadBtn_Click(object sender, RoutedEventArgs e)
                 }
             }
 
+            // API version compatibility check
+            RegistryKey ver_key = Registry.ClassesRoot.OpenSubKey("CLSID\\{"+comType.GUID+"}\\Version");
+            string ver_str = (string)ver_key.GetValue("");
+            string cur_ver = string.Format("{0}.{1}", (int)Image3dAPIVersion.IMAGE3DAPI_VERSION_MAJOR, (int)Image3dAPIVersion.IMAGE3DAPI_VERSION_MINOR);
```

Comment:
It seems like C# doesn't auto-convert enum values to int as C++. Instead, enum value names are converted to strings, which is not what we want here. This was a surprise to me when writing the code.

Failure to cast to int results in `cur_ver = "IMAGE3DAPI_VERSION_MAJOR.IMAGE3DAPI_VERSION_MAJOR"` instead of `cur_ver = "1.1"`

Reply:
Yes, you are right. I did not see that Image3dAPIVersion is an enum.